File: JAVA/HERANÇA/funcionario.java
package JAVA.HERANÇA;


public class Funcionario {
    protected String nome;
    protected double salario;

    public Funcionario(String nome, double salario) {
        this.nome = nome;
        this.salario = salario;
    }

    public void apresentar() {
        System.out.println("Olá, meu nome é " + this.nome + " e meu salário é R$ " + this.salario);
    }
}

// Classe Filha 1
class Gerente extends Funcionario {
    private String setor;

    public Gerente(String nome, double salario, String setor) {
        super(nome, salario); // Reaproveita o construtor da classe mãe
        this.setor = setor;
    }

    public void gerenciar() {
        System.out.println(this.nome + " está gerenciando o setor de " + this.setor + ".");
    }
}

// Classe Filha 2
class Desenvolvedor extends Funcionario {
    private String linguagem;

    public Desenvolvedor(String nome, double salario, String linguagem) {
        super(nome, salario); // Reaproveita o construtor da classe mãe
        this.linguagem = linguagem;
    }

    public void programar() {
        System.out.println(this.nome + " está codando em " + this.linguagem + ".");
    }
}

File: PYTHON/HERANÇA/funcionario.py
class Funcionario:
    def __init__(self,nome,salario):
        self.nome = nome
        self.salario = salario

    def apresentar(self):
        print(f"o meu nome é {self.nome} meu salario é {self.salario}")


class Gerente(Funcionario):
    def __init__(self,nome,salario,setor):
        super().__init__(nome,salario) # Puxa os atributos da classe mãe
        self.setor = setor





    def gerenciar(self):
        print(f"meu nome é {self.nome} sou do setor {self.setor} ganho R${self.salario}")




class Dessenvolvedor(Funcionario):
    def __init__(self, nome, salario, linguagem):
        super().__init__(nome, salario)  # Puxa os atributos da classe mãe
        self.linguagem = linguagem

    def programar(self):
        print(f"{self.nome} está codando em {self.linguagem}.")